coala_langserver/langserver.py

```diff
@@ -110,7 +102,37 @@ def send_diagnostics(self, path, diagnostics):
             'uri': 'file://{0}'.format(path),
             'diagnostics': _diagnostics,
         }
-        self.send_notification('textDocument/publishDiagnostics', params)
+        self._endpoint.notify('textDocument/publishDiagnostics', params=params)
+
+
+@enforce_signature
+def start_tcp_lang_server(handler_class: LangServer, bind_addr, port):
+    # Construct a custom wrapper class around the user's handler_class
+    wrapper_class = type(
+        handler_class.__name__ + 'Handler',
+        (_StreamHandlerWrapper,),
+        {'DELEGATE_CLASS': handler_class}
+    )
+
+    try:
+        server = socketserver.TCPServer((bind_addr, port), wrapper_class)
+    except Exception as e:
+        log('Fatal Exception: {}'.format(e))
+        sys.exit(1)
+    try:
+        log('Serving {} on ({}, {})'.format(
+            handler_class.__name__, bind_addr, port))
+        server.serve_forever()
+    finally:
+        log('Shutting down')
+        server.server_close()
+
+
+@enforce_signature
+def start_io_lang_server(handler_class: LangServer, rstream, wstream):
+    log('Starting {} IO language server'.format(handler_class.__name__))
```

Comment:
new `log` call, also redundant.

Reply:
For the reasons mentioned in the previous change request and maintaining uniformity I think I'll have to retain this log and remove the other redundant log call.

Reply:
ok.